reducers/gotPlayer/peers.js

```diff
@@ -0,0 +1,99 @@
+import { playerPeersActions } from '../../actions';
+import { SORT_ENUM } from '../utility';
+import { combineReducers } from 'redux';
+
+const initialState = {
+  loading: true,
+  error: false,
+  loaded: false,
+  peerList: [],
+  sortState: '',
+  sortField: '',
+  sortFn: f => f,
+};
+
+const peers = (state = initialState, action) => {
+  switch (action.type) {
+    case playerPeersActions.REQUEST:
+      return {
+        ...state,
+        loaded: false,
+        loading: true,
+      };
+    case playerPeersActions.OK:
+      return {
+        ...state,
+        loading: false,
+        loaded: true,
+        peerList: [...action.payload],
+      };
+    case playerPeersActions.ERROR:
+      return {
+        ...state,
+        loading: false,
+        loaded: false,
+        error: true,
+      };
+    case playerPeersActions.SORT:
+      return {
+        ...state,
+        sortState: action.sortField === state.sortField ? SORT_ENUM.next(SORT_ENUM[state.sortState]) : SORT_ENUM[0],
+        sortField: action.sortField,
+        sortFn: action.sortFn,
+      };
+    default:
+      return state;
+  }
+};
+
+
+const byId = (state = {}, action) => {
```

Comment:
I think that I have found a pattern for the reducers which should be the same for everything. So what we can do is factor out a lot of the shared logic and just have some sort of factory function. I'm pretty sure it's possible since changing it is a lot of copy pasting. We can probalbyd o the same for some of the actions and selectors as well, though it may not be worth it for some of the duplication (calling the factory will be just as much code as not calling it).